rules/NamingConventionRule.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\PHPStan\Rules;
+
+use PhpParser\Node;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleErrorBuilder;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Stmt\ClassLike>
+ */
+final class NamingConventionRule implements Rule
```

Comment:
To me, this name is too vague as we might introduce further naming conventions in the future:
```suggestion
final class InterfaceShouldContainSuffix implements Rule
```

Reply:
This class also checks abstract prefix, and trait suffix tho. Unless we want to split it, but from my pov this is all related.

What about `ClassTypeNamingSuffixRule` or `TypeDeclarationNamingRule`?

Reply:
Then if possible, I would personally split this into three rules.

If this can't be the case I vote for `ClassTypeNamingRule` as there is also prefix-related checking going on, so `suffix` in the class name would not fit in my opinion.

Reply:
Of course it can be split, I just dont think it should but w/e.

Reply:
What I meant is I leave this decision to you, to me it's not a must.